storage/integrity_repository.go

```diff
@@ -107,30 +107,24 @@ func (cr *integrityRepository) Delete(ctx context.Context, objectType types.Obje
 	return cr.repository.Delete(ctx, objectType, criteria...)
 }
 
-// InTransaction wraps repository passed in the transaction to also validate integrity
 func (cr *TransactionalIntegrityRepository) InTransaction(ctx context.Context, f func(ctx context.Context, storage Repository) error) error {
-	return cr.repository.InTransaction(ctx, func(ctx context.Context, storage Repository) error {
-		return f(ctx, &integrityRepository{
-			repository:         storage,
-			integrityProcessor: cr.integrityProcessor,
-		})
-	})
+	return cr.repository.InTransaction(ctx, f)
```

Comment:
disabling integrity validation in transaction is not a problem ATM as we don't have transitive integral resources that require manipulation in a transaction opened for the top resource